contracts/Ops.sol

```diff
@@ -1,60 +1,81 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity 0.8.12;
 
-import {Gelatofied} from "./vendor/gelato/Gelatofied.sol";
-import {GelatoBytes} from "./vendor/gelato/GelatoBytes.sol";
 import {
     EnumerableSet
 } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {Gelatofied} from "./vendor/gelato/Gelatofied.sol";
+import {GelatoBytes} from "./vendor/gelato/GelatoBytes.sol";
 import {LibOps} from "./libraries/LibOps.sol";
-import {
-    ITaskTreasuryUpgradable
-} from "./interfaces/ITaskTreasuryUpgradable.sol";
+import {LibLegacyTask} from "./libraries/LibLegacyTask.sol";
 import {IOps} from "./interfaces/IOps.sol";
-import {
-    IOpsProxyFactory
-} from "./vendor/proxy/opsProxy/interfaces/IOpsProxyFactory.sol";
+import {DataTypes} from "./libraries/DataTypes.sol";
+import {Events} from "./libraries/Events.sol";
+import {TaskModules} from "./taskModules/TaskModules.sol";
+import {OpsProxied} from "./vendor/proxy/opsProxy/OpsProxied.sol";
 
 // solhint-disable max-line-length
-// solhint-disable max-states-count
 // solhint-disable not-rely-on-time
 /// @notice Ops enables everyone to communicate to Gelato Bots to monitor and execute certain transactions
 /// @notice ResolverAddresses determine when Gelato should execute and provides bots with
 /// the payload they should use to execute
 /// @notice ExecAddress determine the actual contracts to execute a function on
-contract Ops is Gelatofied, LibOps, IOps {
-    using GelatoBytes for bytes;
+contract Ops is Gelatofied, LibOps, TaskModules, OpsProxied, IOps {
     using EnumerableSet for EnumerableSet.Bytes32Set;
-
-    // solhint-disable const-name-snakecase
-    string public constant version = "4";
-    mapping(bytes32 => address) public taskCreator;
-    mapping(bytes32 => address) public execAddresses;
-    mapping(address => EnumerableSet.Bytes32Set) internal _createdTasks;
-    ITaskTreasuryUpgradable public immutable override taskTreasury;
-    uint256 public fee;
-    address public feeToken;
-    // Appended State
-    mapping(bytes32 => Time) public timedTask;
-    IOpsProxyFactory public immutable override opsProxyFactory;
+    using GelatoBytes for bytes;
 
     constructor(
         address payable _gelato,
-        ITaskTreasuryUpgradable _taskTreasury,
-        IOpsProxyFactory _opsProxyFactory
-    ) Gelatofied(_gelato) {
-        taskTreasury = _taskTreasury;
-        opsProxyFactory = _opsProxyFactory;
+        address _taskTreasury,
+        address _opsProxyFactory
+    ) Gelatofied(_gelato) TaskModules(_taskTreasury, _opsProxyFactory) {}
+
+    // prettier-ignore
+    fallback(bytes calldata _callData) external returns(bytes memory returnData){
```

Comment:
does that work? I would have assumed that you need to use `msg.data` or smth to get the function selector. Did we test this thoroughly?